### ISSUE TITLE
Investigation of some articles missed by SFX integration

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1359]
Ones to look at are here: https://docs.google.com/spreadsheets/d/1aZTdslzp341kNIeSzLYdggOf2YjwrY7IRFVrz5ckVHg/edit#gid=0&fvid=337318064

Shouldn't be done prior to #1356 as they may be related or this may be fixed by it. Same link resolver as that. 

I don't know what's going on in these cases, so would appreciate you investigating and flagging issues that need discussion or making quick fixes where possible.

**Comment by JosephMcArthur** [comment]
> Shouldn't be done prior to #1356 as they may be related or this may be fixed by it. Same link resolver as that.

The uni has since confirmed that due to covid they don't need to worry about print records for release. So, we can ignore these cases for now, unless fixing these is helpful for addressing this.

**Comment by JosephMcArthur** [comment]
Their embeds are at:

https://library.ualberta.ca/sandbox/instant-ill
https://www.library.ualberta.ca/services/interlibrary-loan

They're running the same embed.

I'm going to use https://library.ualberta.ca/sandbox/instant-ill to test.

**Comment by JosephMcArthur** [comment]
Right, I've been updating my last comment like mad, but confirming I'm done for the night.

**Comment by JosephMcArthur** [comment]
Underway & clarified what is actually needed to be done.

**Comment by JosephMcArthur** [comment]
@markmacgillivray can you provide an update here?

**Comment by markmacgillivray** [comment]
I put some specifics in the sheet but here are the general issues:

* crossref title search used to be bad, now it is better. Even with better crossref search we still had a problem with their results not always being right and we had trouble balancing a match without also letting too much wrong stuff through. I've since augmented that with an additional check on the levenshtein distance between the title we give and the title we return, and if within 10% we accept the crossref result. That fixes at least one of the current examples.

* bing has been turned on for dev now, and testing the title lookups on dev does find more metadata for quite a few of them, which should help

* note though that in testing it would appear that sfx only succeeds if we give it at least the article title, journal title or issn, and volume. Without those it does not work. So even if we find a doi but not those particular pieces of metadata there will still be some for which there is nothing else we can do (this shouldn't be too many though, with a DOI or similar ID we usually do get volume info back in the metadata from crossref, epmc, etc)

* about some sfx links going to unexpected places - it's not clear what the problem really is there. I checked a few directly and the sfx XML appears to give the links we are extracting. The rule we use to extract is to find the first getFullTxt link, get the target URL from it, and as long as it does not say "getitnow" in it, then use it. If you're expecting some different link to be used, we'd need to know how to identify that.
